src/Text.Analyzers/Core (Net4)/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -0,0 +1,416 @@
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
+                                                                             description: LocalizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb264492.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+        internal static DiagnosticDescriptor TypeTypeParameterMoreMeaningfulNameRule = new DiagnosticDescriptor(RuleId,
+                                                                             LocalizableTitle,
+                                                                             LocalizableMessageTypeTypeParameterMoreMeaningfulName,
+                                                                             DiagnosticCategory.Naming,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
+                                                                             description: LocalizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb264492.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+        internal static DiagnosticDescriptor MethodTypeParameterMoreMeaningfulNameRule = new DiagnosticDescriptor(RuleId,
+                                                                             LocalizableTitle,
+                                                                             LocalizableMessageMethodTypeParameterMoreMeaningfulName,
+                                                                             DiagnosticCategory.Naming,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
+                                                                             description: LocalizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb264492.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        private SemanticModel _model;
+        private NamingService _namingService;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            AssemblyRule,
```

Comment:
:memo: Only one diagnostic is required here since they all have the same ID

Reply:
No longer applies